lib/web/ui/usercontext.go

```diff
@@ -207,9 +207,22 @@ func NewUserContext(user types.User, userRoles services.RoleSet, features proto.
 	dbServerAccess := newAccess(userRoles, ctx, types.KindDatabaseServer)
 	dbAccess := newAccess(userRoles, ctx, types.KindDatabase)
 	kubeServerAccess := newAccess(userRoles, ctx, types.KindKubeServer)
-	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
 	desktopAccess := newAccess(userRoles, ctx, types.KindWindowsDesktop)
 	cnDiagnosticAccess := newAccess(userRoles, ctx, types.KindConnectionDiagnostic)
+	requestAccess := newAccess(userRoles, ctx, types.KindAccessRequest)
+
+	// If the user has a requestable role or any `search_as_roles`,
+	// they can create access requests for themselves even without an
+	// explicit `allow` rule
+	if len(userRoles.GetAllowedSearchAsRoles()) > 0 || len(accessCapabilities.RequestableRoles) > 0 {
+		requestAccess = access{
+			List:   true,
+			Read:   true,
+			Edit:   true,
+			Create: true,
+			Delete: true,
```

Comment:
Is `Edit` needed, or is that only for approvals?

What about delete?

Reply:
Yes, this should be only `read`, `create` and `list`. Fixed in https://github.com/gravitational/teleport/pull/20209/commits/877f6127cdbe23d52bca52d8f712e6d96161946b